### PR TITLE
Fix existing bugs before moving to next iterations

### PR DESCRIPTION
**Pull request by Anindyadeep**
Here is the status for each of the engines (For Each of the engine, this PR also addes proper versioning inside the requirements, which was causing some problem in reproduction of results)

- [X] AutoAWQ: Need to add `auto-awq` kernel library for making it work
- [X] CTranslate2
- [X] CTransformers 
- [X] DeepSpeed 
- [X] PyTorch Transformers 
- [X] ExllamaV2: Added `tokenizers` library inside requirements
- [X] Optimum Nvidia: Added official image

**Note:** Removing the following engines, since they are not contributing much in the current inference market:

1. Burn 
2. TinyGrad

For AutoGPTQ, and OnnX I will do in a different issue. Since, we need to make a decision on whether we are going for Mistral or not. Because Mistral has [Onnx support too here](https://huggingface.co/microsoft/Mistral-7B-v0.1-onnx)

### CODE REVIEW

File: bench_vllm/setup.sh [hunk 1]
```diff
@@ -44,9 +46,9 @@ install_vllm_cuda() {
         fi
         echo "Installing vllm for CUDA 11.8 with Python version: $PY_VERSION"
         # Download vllm for CUDA 11.8 and specified Python version
-        pip install https://github.com/vllm-project/vllm/releases/download/v0.2.2/vllm-0.2.2+cu118-"$PY_VERSION"-"$PY_VERSION"-manylinux1_x86_64.whl
-        pip install torch --upgrade --index-url https://download.pytorch.org/whl/cu118
-        pip install huggingface-cli
+        "$PYTHON_CMD" -m pip install https://github.com/vllm-project/vllm/releases/download/v0.2.2/vllm-0.2.2+cu118-"$PY_VERSION"-"$PY_VERSION"-manylinux1_x86_64.whl
+        "$PYTHON_CMD" -m pip install torch --upgrade --index-url https://download.pytorch.org/whl/cu118
+        "$PYTHON_CMD" -m pip install huggingface-cli
```
Comment (nsosio) [inline bench_vllm/setup.sh:51]: can we add also here the version of the packages?

Comment (Anindyadeep) [reply bench_vllm/setup.sh:51]: Ah yes, I missed that here, lemme add those

Comment (Anindyadeep) [reply bench_vllm/setup.sh]: Done

Comment (nsosio) [reply bench_vllm/setup.sh]: can you add also torch version?

Comment (Anindyadeep) [reply bench_vllm/setup.sh]: `torch --upgrade --index-url https://download.pytorch.org/whl/cu118`

This one is something I added from [vLLM documentation](https://docs.vllm.ai/en/latest/getting_started/installation.html), so I intentionally did not kept the version, just not to mess up with version issues.

**Comment by nsosio** [comment]
@Anindyadeep one last comment, can you remove the removed engines from [docs/llama2.md.template](https://github.com/premAI-io/benchmarks/blob/main/docs/llama2.md.template)

**Comment by Anindyadeep** [comment]
> @Anindyadeep one last comment, can you remove the removed engines from [docs/llama2.md.template](https://github.com/premAI-io/benchmarks/blob/main/docs/llama2.md.template)?

Done

**Pull request merged**